crypto/rand/rand_lib.c

```diff
@@ -966,4 +1076,60 @@ int RAND_set_seed_source_type(OSSL_LIB_CTX *ctx, const char *seed,
         && random_set_string(&dgbl->seed_propq, propq);
 }
 
-#endif
+int RAND_set1_random_provider(OSSL_LIB_CTX *ctx, OSSL_PROVIDER *prov)
+{
+    RAND_GLOBAL *dgbl = rand_get_global(ctx);
+
+    if (dgbl == NULL)
+        return 0;
+
+    if (prov == NULL) {
+        free_random_provider_name(dgbl);
+        dgbl->random_provider_name = NULL;
```

Comment:
Shouldn't this default back to `random_provider_fips_name` (and if so set `dgbl->rand_provider` to point to the fips provider if it is currently activated)?

Reply:
No, passing NULL for the provider turns this feature off.

Reply:
Hmmm. Then the docs are quite confusing. From the docs:

> Pass NULL for the provider to return to using the built-in in DRBGs and entropy source.

The "return to" here suggested to me that passing NULL was intended to return to the default situation. But the default is to use fips if its available - and only then use the built-in DRBGs. But here you are saying that passing NULL turns this off entirely. That means you can't really get back to the default case (although you could call `RAND_set1_random_provider()` and pass the fips provider - which is almost, but not quite the default situation). Either way I think the docs need some clarification.

Reply:
Good point, return to using was a poor choice of words.  I've amended.